service/matching/taskQueueManager_test.go

```diff
@@ -490,333 +499,266 @@ func TestAddTaskStandby(t *testing.T) {
-			tqm.matchingClient = mockMatchingClient
-		})
-	subTq.Start()
-	require.NoError(t, subTq.WaitUntilInitialized(ctx))
-	for i := 0; i < 10; i++ {
-		_, err := subTq.GetUserData(ctx)
-		require.Error(t, err)
-		time.Sleep(time.Millisecond * 2)
-	}
-}
-
-func TestActivityQueueGetsUserDataFromWorkflowQueue(t *testing.T) {
-	controller := gomock.NewController(t)
-	defer controller.Finish()
-	ctx := context.Background()
-
-	data := mkUserData(1)
-	versionedData := &persistencespb.VersionedTaskQueueUserData{
+	data1 := &persistencespb.VersionedTaskQueueUserData{
 		Version: 1,
-		Data:    data,
-	}
-	asResp := &matchingservice.GetTaskQueueUserDataResponse{
-		TaskQueueHasUserData: true,
-		UserData:             versionedData,
+		Data:    mkUserData(1),
 	}
 
-	tqCfg := defaultTqmTestOpts(controller)
-	tqCfg.tqId.taskType = enumspb.TASK_QUEUE_TYPE_ACTIVITY
-	actTq := mustCreateTestTaskQueueManagerWithConfig(t, controller, tqCfg,
-		func(tqm *taskQueueManagerImpl) {
-			mockMatchingClient := matchingservicemock.NewMockMatchingServiceClient(controller)
-			mockMatchingClient.EXPECT().GetTaskQueueUserData(gomock.Any(), gomock.Any()).
-				Return(asResp, nil).Times(1)
-			tqm.matchingClient = mockMatchingClient
-		})
-	actTq.Start()
-	require.NoError(t, actTq.WaitUntilInitialized(ctx))
-
-	subTqId, err := newTaskQueueIDWithPartition(defaultNamespaceId, defaultRootTqID, enumspb.TASK_QUEUE_TYPE_ACTIVITY, 1)
-	require.NoError(t, err)
-	tqCfg = defaultTqmTestOpts(controller)
-	tqCfg.tqId = subTqId
-	actTqPart := mustCreateTestTaskQueueManagerWithConfig(t, controller, tqCfg,
-		func(tqm *taskQueueManagerImpl) {
-			mockMatchingClient := matchingservicemock.NewMockMatchingServiceClient(controller)
-			mockMatchingClient.EXPECT().GetTaskQueueUserData(gomock.Any(), gomock.Any()).
-				Return(asResp, nil).Times(1)
-			tqm.matchingClient = mockMatchingClient
-		})
-	actTqPart.Start()
-	require.NoError(t, actTqPart.WaitUntilInitialized(ctx))
-
-	actTqData, err := actTq.GetUserData(ctx)
-	require.NoError(t, err)
-	require.Equal(t, versionedData, actTqData)
-	actTqData, err = actTqPart.GetUserData(ctx)
+	tqCfg.matchingClientMock.EXPECT().GetTaskQueueUserData(
+		gomock.Any(),
+		&matchingservice.GetTaskQueueUserDataRequest{
+			NamespaceId:              defaultNamespaceId.String(),
+			TaskQueue:                defaultRootTqID,
+			LastKnownUserDataVersion: 0,
+			WaitNewData:              true,
+		}).
+		Return(&matchingservice.GetTaskQueueUserDataResponse{
+			TaskQueueHasUserData: true,
+			UserData:             data1,
+		}, nil)
+
+	tq := mustCreateTestTaskQueueManagerWithConfig(t, controller, tqCfg)
+	tq.config.GetUserDataMinWaitTime = 10 * time.Second // wait on success
+	tq.Start()
+	require.NoError(t, tq.WaitUntilInitialized(ctx))
+	userData, _, err := tq.GetUserData(ctx)
 	require.NoError(t, err)
-	require.Equal(t, versionedData, actTqData)
-
-	actTq.Stop()
-	actTqPart.Stop()
+	require.Equal(t, data1, userData)
+	tq.Stop()
 }
```

Comment:
Do we have any test for the case where user data is fetched OK, but then we don't see some long-poll update that we expected to see, but things still become eventually consistent?

Reply:
I'm not sure I follow.. you mean like our long polls fail for a while and then eventually we get one that's several versions ahead? That could certainly happen but I'm not sure it's worth a test, there's nothing special in the code about v+1 vs greater

Reply:
Yes, that's what I mean. I'm fine with not having one if it's not meaningfully different.